backend/alternative_flights/alt_flight_search.py

```diff
@@ -0,0 +1,232 @@
+import os
+import json
+from datetime import datetime, timedelta
+import uuid
+from typing import List, Dict
+import numpy as np
+from scipy import stats
+
+class Flight:
+    def __init__(self, departure_airport, flight_data):
+        self.flight_id = str(uuid.uuid4())
+        self.departure_airport = departure_airport
+        self.arrival_airport = flight_data['arrival_airport']
+        self.departure_time = datetime.strptime(flight_data['departureTime'], "%d-%m-%Y %H:%M:%S")
+        self.arrival_time = datetime.strptime(flight_data['arrivalTime'], "%d-%m-%Y %H:%M:%S")
+        self.airline_code = flight_data['airlineCode']
+        self.airline_name = flight_data['airlineName']
+        self.flight_number = flight_data['flightNumber']
+        self.price = flight_data['price']
+        self.duration = float(flight_data['duration'])
+        self.distance = flight_data['distance']
+
+    def to_dict(self):
+        return {
+            'flight_id': self.flight_id,
+            'departure_airport': self.departure_airport,
+            'arrival_airport': self.arrival_airport,
+            'departure_time': self.departure_time.strftime("%d-%m-%Y %H:%M:%S"),
+            'arrival_time': self.arrival_time.strftime("%d-%m-%Y %H:%M:%S"),
+            'airline_code': self.airline_code,
+            'airline_name': self.airline_name,
+            'flight_number': self.flight_number,
+            'price': self.price,
+            'duration': self.duration,
+            'distance': self.distance
+        }
+
+class FlightOption:
+    def __init__(self):
+        self.option_id = str(uuid.uuid4())
```

Comment:
What would be the usage of option id? Do we need to keep this consistent or would it change everything we run this function?

Reply:
This is just random id I'm assigning to each route the DFS function creates. I added this in case we decide to do something with the routes later. No need to keep this consistent, just a reference to the routes.